sqlite3.go

```diff
@@ -1568,9 +1568,11 @@ func (d *SQLiteDriver) Open(dsn string) (driver.Conn, error) {
 		}
 	}
 
-	// Preform Authentication
-	if err := conn.Authenticate(authUser, authPass); err != nil {
-		return nil, err
+	if conn.AuthEnabled() {
```

Comment:
I don't think we can add this check. Now `AuthEnabled()` always returns false even if you use the build tag. Therefore, anyone who had been relying on it just skips auth instead of getting the error.

Reply:
Yes. If the user doesn't need this extension, then the fix is simple - stop passing the build tag they don't need.